src/algorithms/tracking/CKFTracking.cc

```diff
@@ -158,17 +169,99 @@ namespace eicrecon {
+                trajectory.setNMeasurements(trajectoryState.nMeasurements);
+                trajectory.setNStates(trajectoryState.nStates);
+                trajectory.setNOutliers(trajectoryState.nOutliers);
+                trajectory.setNHoles(trajectoryState.nHoles);
+                trajectory.setNSharedHits(trajectoryState.nSharedHits);
+
+                for (const auto& measurementChi2 : trajectoryState.measurementChi2) {
+                    trajectory.addToMeasurementChi2(measurementChi2);
+                }
+
+                for (const auto& outlierChi2 : trajectoryState.outlierChi2) {
+                    trajectory.addToOutlierChi2(outlierChi2);
+                }
+
+                // Get the fitted track parameter
+                //
+                if (multiTrajectory->hasTrackParameters(trackTip)) {
+
+                    const auto& boundParam = multiTrajectory->trackParameters(trackTip);
+                    const auto& parameter  = boundParam.parameters();
+                    const auto& covariance = *boundParam.covariance();
+
+                    edm4eic::MutableTrackParameters pars{
+                        0, // type: track head --> 0
+                        {
+                            static_cast<float>(parameter[Acts::eBoundLoc0]),
+                            static_cast<float>(parameter[Acts::eBoundLoc1])
+                        },
+                        {
+                            static_cast<float>(covariance(Acts::eBoundLoc0, Acts::eBoundLoc0)),
+                            static_cast<float>(covariance(Acts::eBoundLoc1, Acts::eBoundLoc1)),
+                            static_cast<float>(covariance(Acts::eBoundLoc0, Acts::eBoundLoc1))
+                        },
+                        static_cast<float>(parameter[Acts::eBoundTheta]),
+                        static_cast<float>(parameter[Acts::eBoundPhi]),
+                        static_cast<float>(parameter[Acts::eBoundQOverP]),
+                        {
+                            static_cast<float>(covariance(Acts::eBoundTheta, Acts::eBoundTheta)),
+                            static_cast<float>(covariance(Acts::eBoundPhi, Acts::eBoundPhi)),
+                            static_cast<float>(covariance(Acts::eBoundQOverP, Acts::eBoundQOverP)),
+                            static_cast<float>(covariance(Acts::eBoundTheta, Acts::eBoundPhi)),
+                            static_cast<float>(covariance(Acts::eBoundTheta, Acts::eBoundQOverP)),
+                            static_cast<float>(covariance(Acts::eBoundPhi, Acts::eBoundQOverP))
+                        },
+                        static_cast<float>(parameter[Acts::eBoundTime]),
+                        sqrt(static_cast<float>(covariance(Acts::eBoundTime, Acts::eBoundTime))),
+                        static_cast<float>(boundParam.charge())};
+
+                    track_parameters->push_back(pars);
+                    trajectory.addToTrackParameters(pars);
```

Comment:
Fun fact: This is not compatible with EDM4eic 2.0.0. I suppose, it's not practical to support it, but want to point it out.

Reply:
Right, I forgot to update the minimum. The whole reason for 2.1.0 was to allow this...

Reply:
https://github.com/eic/EICrecon/pull/885